app/components/themed/ThemedTextActivityIndicator.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { View } from 'react-native'
+import { tailwind } from '@tailwind'
+import { ThemedActivityIndicator } from './ThemedActivityIndicator'
+import { ThemedText } from './ThemedText'
+import { ThemedProps } from '.'
+
```

Comment:
I prefer adding it on the button itself. No need to have this element since it's always displayed in a button. It would be better to just pass a "loading" param in the button (that triggers the display for ActivityIndicator) and dynamic text (e.g, converting.. convert etc.)